### PR TITLE
feat: add `delete` functionality to list feature

### LINKED ISSUE #1731: feat: add `Delete` feature to list
Will it be out of scope if I add a `Delete` flow for the list?

https://github.com/open-sauced/app/assets/62995161/76b049ed-d2b0-4676-a427-c0cf8826c412

Similar to what was done with insights pages

_Originally posted by @OgDev-01 in https://github.com/open-sauced/app/issues/1698#issuecomment-1728578103_

### PR DESCRIPTION
**Pull request by OgDev-01**
## Description
This PR
- adds the ability to delete a list
- fixes some minor ui bugs

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
closes #1731 

## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/app/assets/62995161/c0f365fe-a7b2-4601-a0ce-11c7d1196a8f

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,105 +7,92 @@
 
 ## [1.66.0-beta.3](https://github.com/open-sauced/insights/compare/v1.66.0-beta.2...v1.66.0-beta.3) (2023-09-22)
 
-
 ### 🐛 Bug Fixes
 
-* fix search bar bug ([#1735](https://github.com/open-sauced/insights/issues/1735)) ([d16e93f](https://github.com/open-sauced/insights/commit/d16e93fc06585f0db735abce9394dd55fc1ce633))
+- fix search bar bug ([#1735](https://github.com/open-sauced/insights/issues/1735)) ([d16e93f](https://github.com/open-sauced/insights/commit/d16e93fc06585f0db735abce9394dd55fc1ce633))
```
Comment (nickytonline) [inline CHANGELOG.md:12]: Prettier in action?

File: pages/hub/lists/index.tsx [hunk 2]
```diff
@@ -66,6 +115,59 @@ const ListsHub: WithPageLayout = () => {
           goToPage
         />
       </div>
+
+      <Dialog open={isDeleteOpen}>
```
Comment (nickytonline) [inline pages/hub/lists/index.tsx:119]: This doesn't block the PR, but I've seen the Dialog component being used in several places where we add the `<DialogContent />`, <DialogTitle />` etc.

This can be abstracted into a component that would look something like this.

```jsx
import { Dialog as RadixDialog, DialogContent, DialogDescription, DialogTitle } from "@radix-ui/react-dialog";
import { DialogCloseButton, DialogHeader } from "components/molecules/Dialog/dialog";

// define the types for the props for the DialogyComponent
interface DialogProps {
  children: React.ReactNode;
  title: string;
  description?: string;
  onClose: () => void;
  onOpen: boolean;
  onOpenChange: (open: boolean) => void;
}

export function Dialog({ children, title, description, onClose, onOpen: isOpen, onOpenChange }: DialogProps) {
  return (
    <RadixDialog open={isOpen} onOpenChange={onOpenChange}>
      <DialogContent className="p-4">
        <DialogHeader>
          <DialogTitle>{title}</DialogTitle>
          {description && <DialogDescription className="font-normal">{description}</DialogDescription>}
        </DialogHeader>
        {children}
        <DialogCloseButton onClick={onClose} />
      </DialogContent>
    </RadixDialog>
  );
}
```


I'll create an issue for it.

Comment (nickytonline) [reply pages/hub/lists/index.tsx:119]: I created https://github.com/open-sauced/app/issues/1741

File: layouts/hub.tsx [hunk 3]
```diff
@@ -17,7 +17,10 @@ import TopNav from "../components/organisms/TopNav/top-nav";
 const HubLayout = ({ children }: { children: React.ReactNode }) => {
   const { onboarded } = useSession();
   const { user } = useSupabaseAuth();
-  const navLinks = [{ name: "Insights", href: "/hub/insights" }];
+  const navLinks = [
+    { name: "Insights", href: "/hub/insights" },
+    { name: "Lists", href: "/hub/lists" },
```
Comment (brandonroberts) [inline layouts/hub.tsx:22]: Don't include this yet

```suggestion

```

File: pages/hub/lists/index.tsx [hunk 4]
```diff
@@ -9,19 +9,68 @@ import PaginationResults from "components/molecules/PaginationResults/pagination
 import Pagination from "components/molecules/Pagination/pagination";
 import SkeletonWrapper from "components/atoms/SkeletonLoader/skeleton-wrapper";
 import Title from "components/atoms/Typography/title";
+import Text from "components/atoms/Typography/text";
 import ListCard from "components/molecules/ListCard/list-card";
 
 import { useFetchAllLists } from "lib/hooks/useList";
+import { Dialog, DialogContent, DialogTitle } from "components/molecules/Dialog/dialog";
```
Comment (brandonroberts) [inline pages/hub/lists/index.tsx:16]: nit: (not blocking) move the `components` into the group above

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: pages/hub/lists/index.tsx [hunk 5]
```diff
@@ -66,6 +115,59 @@ const ListsHub: WithPageLayout = () => {
...
+          </Text>
+          <Text>
+            If you have data on this list that your team is using it would be difficult for your team to get access to
+            track your project.
+          </Text>
+          <Text>
+            <span className="font-bold text-light-slate-12">This action cannot be undone</span>
+          </Text>
+          <Text>
+            Type <span className="font-bold text-light-red-10">{listNameToDelete}</span> in all caps to confirm
```
[the remark this answers is not in the document]
Comment (jpmcb) [reply pages/hub/lists/index.tsx:136]: I'm also having a hard time reviewing this since I'm running into https://github.com/open-sauced/app/issues/1743 on the Netlify deploy preview

File: CHANGELOG.md (hunk 1 above)
Comment (OgDev-01) [reply CHANGELOG.md:12]: yeah 🥹

File: pages/hub/lists/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (OgDev-01) [reply pages/hub/lists/index.tsx:136]: > "in all caps to confirm"
Is this true? seems you can just delete without all caps.

totally missed this 🥲. updating the copy shortly

File: pages/hub/lists/index.tsx [hunk 6]
```diff
@@ -66,6 +116,59 @@ const ListsHub: WithPageLayout = () => {
...
+          <DialogTitle>
+            <Title level={3}>Delete List</Title>
+          </DialogTitle>
+
+          <Text>
+            Are you sure you want to delete <span className="font-bold text-light-slate-12">{listNameToDelete}</span>?
+          </Text>
+          <Text>
+            If you have data on this list that your team is using it would be difficult for your team to get access to
+            track your project.
```
Comment (jpmcb) [inline pages/hub/lists/index.tsx:131]: I think the copy here could be more clear. Maybe something simple like?

```suggestion
            If you have data on this list that your team is using, they will loose access
```

**Review by jpmcb — APPROVED** [review]
Looking good! 👍🏼

File: next-types.d.ts [hunk 7]
```diff
@@ -50,33 +50,33 @@ interface DbRecommendedInsightsRepo {
...
-  readonly is_onboarded: boolean;
-  readonly is_waitlisted: boolean;
-  readonly role: number;
-  readonly created_at: string;
-  readonly updated_at: string;
-  readonly languages: Object;
-  readonly public_gists: number;
-  readonly public_repos: number;
-  readonly receive_collaborations: boolean;
+  readonly user_avatar_url: string;
```
Comment (brandonroberts) [inline next-types.d.ts:53]: Revert these changes

File: pages/hub/lists/new.tsx [hunk 8]
```diff
@@ -43,9 +43,9 @@ const NewListCreationPage: WithPageLayout = () => {
     ? data.length > 0 &&
       data.map((contributor) => {
         return {
-          author_login: contributor.login,
-          updated_at: contributor.updated_at,
-          user_id: contributor.id,
+          author_login: contributor.user_login,
```
Comment (brandonroberts) [inline pages/hub/lists/new.tsx:46]: Revert these changes

**Review by brandonroberts — CHANGES_REQUESTED** [review]
